code/Control/UserDefinedFormController.php

```diff
@@ -511,6 +533,250 @@ public function finished()
+        if (!$editableFormField instanceof EditableFileField) {
+            $editableFormField->setClassName(EditableFileField::class);
+        }
+        $editableFileField = $editableFormField;
+
+        if (!$editableFileField->canView()) {
+            throw new PermissionFailureException();
+        }
+
+        // check if we're creating a new folder or using an existing folder
+        $option = $request->requestVar('FolderOptions');
+        if ($option == 'existing') {
+            // set existing folder
+            $folderID = $request->requestVar('FolderID');
+            $folder = Folder::get()->byID($folderID);
+        } else {
+            // create the folder
+            $createFolder = $this->config()->get('form_submissions_folder') . '/' . $request->requestVar('CreateFolder');
+            $formSubmissionsFolderExists = !!Folder::find($this->config()->get('form_submissions_folder'));
+            $folder = Folder::find_or_make($createFolder);
+
+            // If this is the first time we create the form submission folder
+            if (!$formSubmissionsFolderExists) {
+                // ensure the FormSubmissions folder is only accessible to Administrators
+                $formSubmissionsFolder = Folder::find($this->config()->get('form_submissions_folder'));
+                $formSubmissionsFolder->CanViewType = InheritedPermissions::ONLY_THESE_USERS;
+                $formSubmissionsFolder->ViewerGroups()->removeAll();
+                $formSubmissionsFolder->ViewerGroups()->add(Group::get_one(Group::class, ['Code' => 'administrators']));
+                $formSubmissionsFolder->write();
+            }
+        }
+
+        // assign the folder
+        $editableFileField->Folder = $folder;
+        $editableFileField->write();
+
+        // respond
+        $response = new HTTPResponse(json_encode([]));
+        $response->addHeader('Content-Type', 'application/json');
+        return $response;
+    }
+
+    protected static function getFolderPermissionTuple(Folder $folder = null)
+    {
+        $viewersOptionsField = [
+            'Inherit' => _t(__CLASS__.'.INHERIT', 'Visibility for this folder is inherited from the parent folder'),
+            'Anyone' => _t(__CLASS__.'.ANYONE', 'Uploads will be visible to anyone'),
+            'LoggedInUsers' => _t(__CLASS__.'.LOGGED_IN', 'Restricted access, uploads will be visible to logged-in users'),
+            'OnlyTheseUsers' => _t(__CLASS__.'.ONLY_GROUPS', 'Restricted access, uploads will be visible to the following groups: ')
+        ];
+
+        if (!$folder) {
+            return [
+                'CanViewType' => '',
+                'ViewerGroups' => [],
+            ];
+        }
+
+        return [
+            'CanViewType' => $viewersOptionsField[$folder->CanViewType],
+            'ViewerGroups' => $folder->ViewerGroups()->column('Title'),
+        ];
+    }
+
+    protected static function getFolderPermissionString(Folder $folder = null)
```

Comment:
Make private unless there's a good reason to make it protected.

Why is it static.

Reply:
Add PHP doc